src/structures/paging/mapper/recursive_page_table.rs

```diff
@@ -110,12 +112,18 @@ impl<'a> RecursivePageTable<'a> {
 
             if entry.is_unused() {
                 if let Some(frame) = allocator.allocate_frame() {
-                    entry.set_frame(frame.frame(), Flags::PRESENT | Flags::WRITABLE);
+                    entry.set_frame(
+                        frame.frame(),
+                        Flags::PRESENT | Flags::WRITABLE | insert_flags,
```

Comment:
We should document that the `PRESENT` and `WRITABLE` flags are always set for higher level page table entries because the recursive page table approach requires it. Preferably we both add a code comment here and also add it to the documentation of the `RecursivePageTable` type so that users see it too.